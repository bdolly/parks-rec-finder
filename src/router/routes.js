import pprfLeafletMap from '@/components/map/pprf-leaflet-map'
// import pprfSidebar from '@/components/pprf-sidebar'
import pprfSidebarSearchContainer from '@/containers/pprf-sidebar-search-container'
import pprfSidebarEntityTaxoTermsContainer from '@/containers/pprf-sidebar-entity-taxo-terms-container'
import pprfSidebarEntityTaxoContainer from '@/containers/pprf-sidebar-entity-taxo-container'

const routes = [
  {
    path: '/', redirect: '/programs'
  },
  {
    path: '/search',
    name: 'Search',
    components: {
      'map': pprfLeafletMap,
      'sidebar': pprfSidebarSearchContainer
    }
  },
  {
    path: '/:entityType/:entityTerm+',
    name: 'Taxonxomy Term Items',
    components: {
      'map': pprfLeafletMap,
      'sidebar': pprfSidebarEntityTaxoContainer
    },
    props: { default: true, sidebar: true }
  },
  {
    path: '/:entityType',
    name: 'Taxonomy Terms',
    components: {
      'map': pprfLeafletMap,
      'sidebar': pprfSidebarEntityTaxoTermsContainer
    },
    props: { default: true, sidebar: true }
  }
]
export default routes
