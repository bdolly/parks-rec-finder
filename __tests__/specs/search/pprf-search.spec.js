import { mount, shallow, createLocalVue } from 'vue-test-utils'
import { createRenderer } from 'vue-server-renderer'
import pprfSearch from '@/components/search/pprf-search'

import Vuex from 'vuex'
import VueRouter from 'vue-router'
import routes from '@/router/routes'

const localVue = createLocalVue()
localVue.use(Vuex)
localVue.use(VueRouter)

const router = new VueRouter({ routes })

describe('search/pprf-search.vue', () => {
  let SearchComponent

  beforeEach(() => {
    SearchComponent = mount(pprfSearch, {localVue, router })
  })

  it('sanity check', () =>{
    expect(true).toBe(true)
    expect(SearchComponent.vm.$route).toBeInstanceOf(Object)
  })

  it('mathches snapshot', () => {
    const renderer = createRenderer()
    renderer.renderToString(SearchComponent.vm, (err, str) => {
      if (err) throw new Error(err)
        expect(str).toMatchSnapshot()
    })
  })
  it('renders a form element', () => {
    expect(SearchComponent.is('form')).toBe(true)
  })
  it('has two text input elements', () => {
    expect(SearchComponent.findAll('input[type="text"]').length).toBe(2)
  })
  it('has a button element that is disabled by default', () => {
     expect(SearchComponent.contains('button[disabled="disabled"]')).toBe(true)
  })
  it('should update the isDisabled prop on text field input', () => {
    let spy = spyOn(console, 'log')
    SearchComponent.vm.onFreetextInput('foobar')
    SearchComponent.vm.$nextTick(() => {
      expect(spy).toHaveBeenCalled()
      expect(SearchComponent.vm.$refs.onFreetextInput.isDirty).toBe(true)
      expect(SearchComponent.vm.isDisabled).toBe(false)
    })
  })
  describe('onAddressInput:', () => {
    it(' VALID zipcode input should ONLY populate the search fields zip state', () => {
      SearchComponent.vm.onAddressInput(19146)
      expect(SearchComponent.vm.search.fields.zip).toBe(19146)
      expect(SearchComponent.vm.search.fields.address).toBe(null)
    })
    it('address input should ONLY populate the search fields address state', () => {
      SearchComponent.vm.onAddressInput('1234 Market St')
      expect(SearchComponent.vm.search.fields.zip).toBe(null)
      expect(SearchComponent.vm.search.fields.address).toBe('1234 Market St')
    })
  })
  describe('onSubmit', () => {
    let actions
    let state
    let store
    beforeEach(() => {
      state = {
        search: {
          fields: {
            freetext: null,
            address: null,
            zip: null
          }
        }
      }
      actions = {
        submitSearch: jest.fn()
      }
      store = new Vuex.Store({
        state,
        actions
      })

    })
    it('calls store action "submitSearch" on form submission', () => {
      const wrapper = mount(pprfSearch, {store, localVue, router})
      wrapper.setMethods({updateSearchRouteParams: jest.fn()})
      wrapper.setData({
        search: {
          fields: {
            freetext: 'Basketball',
            address: '1234 Market st',
            zip: 19147
          }
        }
      })
      wrapper.trigger('submit')
      expect(wrapper.vm.search.fields.freetext).toBe('Basketball')
      expect(actions.submitSearch).toHaveBeenCalled()
    })
  })

  afterEach(() => {
    SearchComponent.destroy()
  })
})
