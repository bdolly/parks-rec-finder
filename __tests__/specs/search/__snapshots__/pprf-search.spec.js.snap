// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`search/pprf-search.vue mathches snapshot 1`] = `
<form class="pprf-search form">
    <div class="field field--freetext"><label id="freetext__label" for="freetext__input"><!----></label>
        <!----><input type="text" id="freetext__input" name="freetext" placeholder="SEARCH BY ACTIVITY TYPE OR LOCATION NAME" tabindex="0" class="form__input form__input--freetext"></div>
    <div class="field field--address field--w-icon"><label id="address__label" for="address__input"><!----></label>
        <!----><input type="text" id="address__input" name="address" placeholder="ADDRESS OR ZIPCODE" tabindex="0" class="form__input form__input--address"></div> <button disabled="disabled" id="pprf-search__submit" tabindex="3" class="input input__submit pprf-btn button button--disabled"><div class="valign"><div class="button-label valign-cell"></div></div></button></form>
`;
